### ISSUE TITLE
post request is hang

### ISSUE THREAD

**Issue by jebai0521** [strongloop/loopback-gateway#14]
get request is ok, but post request is hang

**Comment by raymondfeng** [comment]
What request did you send? Please provide more information.

**Comment by jebai0521** [comment]
@raymondfeng  i am sorry , content-type setting was not correct, the module no problem, thanks

**Comment by jebai0521** [comment]
for example, i create a demo-api-server

**Comment by jebai0521** [comment]
hi ,sorry for reopen the issue,i found when i use oauth2 sdk v1.0.0, all request to the gateway is ok, but if i use oauth2 module version after v2.0.0, all the request was hang on gatway except get request.

**Comment by raymondfeng** [comment]
What do you mean oauth2 sdk 1.0.0? Is it loopback-component-oauth2?

**Comment by jebai0521** [comment]
yes, oauth2 i mentioned was loopback-component-oauth2, i notice that you modify the package.json 3days ago, update

**Comment by raymondfeng** [comment]
Can you post your http request that causes the hang?

**Comment by jebai0521** [comment]
first i  run demo-api-server and gateway.

then i do some operation to call the /api/note in browser, it was success.

url was
`https://localhost:3001/api/notes?access_token=xOXm6QikxWIRKRmlXIaZZATJWbMmwZA0`

result was 
```
[
    {
        title: "bbb",
        content: "ccc",
        id: "54b3545bf3ffbe3a157dc2bb"
    },
    {
        title: "cc",
        content: "dd",
        id: "54b366f8e0633bc702b0bda0"
    },
    {
        title: "aaa",
        content: "bbb",
        id: "54cb32c0040b5c4a5e669093"
    }
]
```

then i use the url from browser to [CocoaRestClient](https://code.google.com/p/cocoa-rest-client/ "CocoaRestClient"), modify verb to post, content type to json, put some properties..

the result show
`Connection to https://localhost:3001/api/notes?access_token=xOXm6QikxWIRKRmlXIaZZATJWbMmwZA0 failed.`.


i log in proxy middleware, 

```
  proxy.on('error', function(err, req, res) {
    var proxyErr = new Error('Proxy error');
    proxyErr.url = req.url;
    proxyErr.target = req.target;
    proxyErr.cause = err;
    console.log("error: " + JSON.stringify(proxyErr));
    res.status(500).send(proxyErr);
  });
```

it show : 
```
error: {"url":"/api/notes?access_token=N9nNWQUPAh87WKNdzC5xJVtEfQh0BIBG","target":"http://localhost:3002","cause":{"code":"ECONNRESET"}}
```

**Comment by jebai0521** [comment]
hi @raymondfeng , was there any wrong.

**Comment by raymondfeng** [comment]
It's a bug in `loopback-component-oauth2` where the `json` parser is triggered for the `/api/notes` path. The body parser consumes the req stream and interferes with http-proxy which pipes the incoming http req into the other server. The fix is released in loopback-component-oauth2@2.0.0-beta4. Please upgrade.
